File: src/org/webfabric/sitemesh/ServletPath.java
package org.webfabric.sitemesh;

import javax.servlet.http.HttpServletRequest;

public class ServletPath implements Path {
    private final String value;

    public ServletPath(HttpServletRequest request) {
        this.value = request.getServletPath();
    }

    public ServletPath(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: src/org/webfabric/servlet/ServletPath.scala
package org.webfabric.servlet;

import org.webfabric.io.{HierarchicalPath}
import javax.servlet.http.HttpServletRequest;

class ServletPath(v: String) extends HierarchicalPath(v)

object ServletPath {
  def apply(request: HttpServletRequest):ServletPath = {
    return new ServletPath(request.getServletPath());
  }
}